client/adjudicate.go

```diff
@@ -231,3 +244,102 @@ func (c *Channel) setMachinePhase(ctx context.Context, e channel.AdjudicatorEven
 
 	return
 }
+
+type mutexList []*sync.Mutex
+
+func (a mutexList) Unlock() {
+	for _, m := range a {
+		m.Unlock()
+	}
+}
+
+// tryLockRecursive tries to lock the channel and all of its sub-channels.
+// It returns a list of all the mutexes that have been locked.
+func (c *Channel) tryLockRecursive(ctx context.Context) (l mutexList, err error) {
+	l = mutexList{}
+	f := func(c *Channel) error {
+		if !c.machMtx.TryLockCtx(ctx) {
+			return errors.Errorf("locking machine mutex in time: %v", ctx.Err())
+		}
+		l = append(l, &c.machMtx)
+		return nil
+	}
+
+	err = f(c)
+	if err != nil {
+		return
+	}
+
+	err = c.applyToSubChannelsRecursive(f)
+	return
+}
+
+// applyToSubChannelsRecursive applies the function to all sub-channels recursively.
+// Assumes sub-channels are locked.
```

Comment:
You use it for locking the sub-channels as well, so i think this comment is not correct.